### DISCUSSION TITLE
rector php install PhpStorm or Visual Studio Code

### DISCUSSION THREAD

**Discussion by MarcoGoeli** [rectorphp/rector#7683] [Q&A] [answered]
Hello

A specific question. I came across rector php because you can update php code to version 8. Now my problem is how do I install this in PhpStorm or Visual Studio Code.
Can someone explain this to me step by step. Thanks.

**Comment by samsonasik** [comment]
iirc, phpstorm only provide command shortcut, you can install it via composer, you can follow the doc

https://getrector.org/documentation

After installed, the phpstorm shortcut command will work.

**Comment by MarcoGoeli** [comment]
In PhpStorm I can do this under Tools/Composer/Install. What do I choose from the menu there? I'm new and not very familiar with PhpStorm.
![composer phpstorm](https://user-images.githubusercontent.com/121103714/208676219-64cdfcf6-39e0-45f1-baa4-ddf5fd6658e3.jpg)

**Comment by samsonasik** [comment, marked as answer]
You need to install composer first:

https://getcomposer.org/

if you are not familiar with composer, you can read the composer documentation https://getcomposer.org/doc/00-intro.md

then, install phpstorm

then, install rector on your project
then, you can see the menu under more -> run -> debug, on right click directory:

![Screen Shot 2022-12-20 at 20 26 33](https://user-images.githubusercontent.com/459648/208677885-4ce00c30-027d-4d23-a4cf-13626b818305.png)

**Comment by MarcoGoeli** [comment]
Okay thanks. I installed PHP8.2 on Windows. Composer installed. Installed PhpStorm and then rector php. However, there are a few reports.

_PS C:\Users\marco\PhpstormProjects\untitled1> composer require rector/rector --dev
Info from https://repo.packagist.org: #StandWithUkraine
./composer.json has been created
Running composer update rector/rector
Loading composer repositories with package information
Updating dependencies
Lock file operations: 2 installs, 0 updates, 0 removals
  - Locking phpstan/phpstan (1.9.4)
  - Locking rector/rector (0.15.1)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 2 installs, 0 updates, 0 removals
    Failed to download phpstan/phpstan from dist: The zip extension and unzip/7z commands are both missing, skipping.
The php.ini used by your command-line PHP is: C:\Program Files\php-8.2.0\php.ini                                                                                                                                                        
    Now trying to download from source

In GitDownloader.php line 82:
                                                            
  git was not found in your PATH, skipping source download  
                                                            

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--update-no-dev] [-w|-
-update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>...]_

**Reply by samsonasik** [reply]
Looking at you composer command ouput, that means you need to install/enable **zip** or **unzip/7z** php extension. 

You can google for that to lookup the solution based on your platform:

Type in **Google** search: `Failed to download phpstan/phpstan from dist: The zip extension and unzip/7z commands are both missing, skipping`.

**Comment by MarcoGoeli** [comment]
I simply installed Php 8 on Windows 11. What do you mean by platform? When I search for the error in Google I find the following code:

`(sudo) yum install zip unzip php-zip
(sudo) apt install zip unzip php-zip

sudo apt-get install zip unzip php8.0-zip

sudo apt-get install zip unzip php7.0-zip`

No idea...

**Reply by samsonasik** [reply]
Yes, you can search on specific OS to do, documentation possibly help https://www.php.net/manual/en/install.pecl.windows.php

You can't learn all things in 1 night, just be slow :), keep up 💪

**Comment by MarcoGoeli** [comment]
Thank you, I am searching and slowly learning.

**Comment by MarcoGoeli** [comment]
Rector cannot be installed under PhpStorm because it is already installed as a plugin (Rector Support) from version 2022.3. Now maybe someone knows how I use it now? The plugin is installed under C:\Program Files\JetBrains\PhpStorm 2022.3\plugins\rector\.

Is this correct that I can create a rector application under Run/Debug Configurations?
1. Assign names
2. Tick "Allow multiple instances"
3. Directory to process plugin path?
4. Configuration file created rector.php
5. Before launch "Run Another Configuration"

![phpstorm](https://user-images.githubusercontent.com/121103714/214009444-3bb6dcfd-1413-4154-98ba-527725d1cbfb.jpg)
